engine-tests/src/benches/res/StandardPrecompiles.sol

```diff
@@ -150,12 +175,19 @@ contract StandardPrecompiles {
     return output;
   }
 
-  function modexp(uint256 base, uint256 exponent, uint256 modulus) private view returns (uint256 output) {
+  function modexp(
```

Comment:
I don't think this function has been modified quite right. I think probably the second argument to `add` in the various `mstore` calls also needs to be modified to account for the changed length.

But rather than mess around with pointer arithmetic in Solidity, I think it would be better to add the new integration tests as separate Rust tests. These tests could call the precompile directly with some constructed input similarly to how the [ecrecover tests](https://github.com/aurora-is-near/aurora-engine/blob/develop/engine-tests/src/tests/ecrecover.rs) are done.